src/FsCodec.SystemTextJson/TypeSafeEnumConverter.fs

```diff
@@ -0,0 +1,52 @@
+﻿namespace FsCodec.SystemTextJson
+
+open System
+open System.Collections.Generic
+open System.Text.Json
+
+/// Utilities for working with DUs where none of the cases have a value
+module TypeSafeEnum =
+
+    let private _isTypeSafeEnum (t : Type) =
+        Union.tryGetUnion t
+        |> Option.exists (fun u -> u.cases |> Seq.forall (fun case -> case.GetFields().Length = 0))
+    let isTypeSafeEnum = memoize _isTypeSafeEnum
+
+    let tryParseT (t : Type) (str : string) =
```

Comment:
Do you think it's worth it to memoize this function (and the others, like `toString`, etc) as well? While `Union.tryGetUnion` is memoized, these functions do call constructors, tag readers, etc each time which are relatively expensive operations - especially when taken in context of JSON documents which could be coming in at 1000s / sec.

Reply:
I'm not averse to that happening, but would prefer to have this PR stick to the task for now
In practice, caching the lookup in the Converter might ultimately give a good solution too
Also relevant is that there is another converter (UnionConverter) which will bring some more Union.* methods into play - I'd like to do an aggressive cleanup/refactor post that point before we risk Perf-Induced-Design-Damage 😈

Reply:
(I can't imagine this not ending up being the right answer; tbh I'm also being just a touch lazy!)

Reply:
should probably also be applied to original; upon considering, opting to defer for now